pyanaconda/modules/payloads/payload/rpm_ostree/installation.py

```diff
@@ -311,6 +327,63 @@ def run(self):
         return self._internal_mounts
 
 
+class PrepareBootcMountTargetsTask(PrepareMountTargetsTaskBase):
+    """Task to prepare Bootc mount targets.
+    """
+
+    @property
+    def name(self):
+        return "Prepare Bootc mount targets"
+
+    def _handle_api_mount_points(self):
+        """Handle API mount points
+
+        Explicitly do API (sysfs, tmpfs,...) bind mounts from host to sysroot for bootc.
+        """
+        for path in ("/proc", "/sys"):
+            sysroot_path = self._sysroot + path
+            safe_exec_program("mount", ["--bind", path, sysroot_path])
+            self._internal_mounts.append(sysroot_path)
+
+    def _handle_var_mount_point(self, existing_mount_points):
+        """Handle /var mount point for bootc.
+
+        For bootc, we bind mount /var similar to OSTree, then create /var/roothome and /var/home
+        in the mounted /var.
+
+        :param [] existing_mount_points: a list of existing mount points
+        """
+        # Bind mount /var using parent class method
+        super()._handle_var_mount_point(existing_mount_points)
+
+        # Create /var/roothome and /var/home after bind mount so they exist in the mounted /var
+        var_sysroot = self._sysroot + "/var"
+        var_roothome = var_sysroot + "/roothome"
+        var_home = var_sysroot + "/home"
+        os.makedirs(var_roothome, mode=0o755, exist_ok=True)
+        os.makedirs(var_home, mode=0o755, exist_ok=True)
+
+    def run(self):
+        """Run the task.
+
+        :return: list of bindmounts created for internal use
+        :rtype: list of str
+        """
+        device_tree = STORAGE.get_proxy(DEVICE_TREE)
+        mount_points = device_tree.GetMountPoints()
+
+        # Remount sysroot and sysimage as read-write so we can set up API mount points
+        safe_exec_program("mount", ["-o", "remount,rw", self._sysroot])
+        safe_exec_program("mount", ["-o", "remount,rw", self._physroot])
+
+        # Handle API mount points - bind mount from host to sysroot
+        # These bind mount the host's API directories to the deployment
+        self._handle_api_mount_points()
+
+        # Handle /var mount point - creates /var/roothome and /var/home, then bind mounts from host
```

Comment:
Is this comment correct? I think the order is bind, then create.

Reply:
Indeed, will fix thanks. Also the bind mounts are nto from host this is obsolete, I will adress.